lib/jasmine/jasmine.ts

```diff
@@ -17,10 +12,14 @@ const _global = typeof window == 'undefined' ? global : window;
 // it. We override the `clearStack` method which forces jasmine to always
 // drain the stack before next test gets executed.
 (<any>jasmine).QueueRunner = (function (SuperQueueRunner) {
+  const originalZone = Zone.current;
   // Subclass the `QueueRunner` and override the `clearStack` method.
 
   function alwaysClearStack(fn) {
-    _global[SET_TIMEOUT](fn, 0);
+    const zone: Zone = Zone.current.getZone('JasmineClearStackZone')
+        || Zone.current.getZone('ProxyZoneSpec')
+        || originalZone;
+    zone.scheduleMicroTask('jasmineCleanStack', fn);
```

Comment:
isn't this going to result in StackOverflowErrors since we are not clearing the stack?

I believe that jasmine calls the "clearStack" method only once in a while when it actually needs to clear the stack.